### PR TITLE
OSDOCS-2285: Added AWS PrivateLink info to configure DNS forwarding.

### PR DESCRIPTION
**Pull request by jonquilwilliams**
JIRA: https://issues.redhat.com/browse/OSDOCS-2285

Previews: 

https://deploy-preview-35314--osdocs.netlify.app/openshift-rosa/latest/rosa_getting_started/rosa-aws-privatelink-creating-cluster?utm_source=github&utm_campaign=bot_dp#rosa-aws-privatelink-create-cluster.adoc_rosa-getting-started

https://deploy-preview-35314--osdocs.netlify.app/openshift-rosa/latest/rosa_architecture/rosa-architecture-models.html#osd-aws-privatelink-architecture.adoc_rosa-architecture-models

Merge to `dedicated-4` branch. No milestones and no CP needed.

### CODE REVIEW

**Comment by yuwang-RH** [comment]
LGTM

File: modules/osd-aws-privatelink-config-dns-forwarding.adoc [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// Module included in the following assemblies:
+//
+// * rosa_getting_started/rosa-aws-privatelink-creating-cluster.adoc
+
+[id="osd-aws-privatelink-config-dns-forwarding_{context}"]
+= Configuring AWS PrivateLink DNS forwarding
+
+With AWS PrivateLink clusters, only a private hosted zone is created in Route53. With a private hosted zone, records within the zone are resolvable only from within the VPC to which it is assigned.
```
Comment (bmcelvee) [inline modules/osd-aws-privatelink-config-dns-forwarding.adoc:8]: s/Route53/Route 53

File: modules/osd-aws-privatelink-config-dns-forwarding.adoc [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+For more information about private hosted zones, see link:https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/hosted-zones-private.html[AWS private hosted zones documentation].
+
+.Prerequisites
+
+* Your corporate network or other VPC has connectivity
+* Ports UDP/53 and TCP/53 ARE enabled across your networks to allow for DNS queries
```
Comment (bmcelvee) [inline modules/osd-aws-privatelink-config-dns-forwarding.adoc:14]: For searchability I would recommend and optional edit here: 

UDP port 53 and TCP port 53 are enabled across your networks to allow for DNS queries

File: modules/osd-aws-privatelink-config-dns-forwarding.adoc [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+* You have created an AWS PrivateLink cluster using {product-title}
+
+.Procedure
+
+. To allow for records such as `api.<cluster_domain>`` and ``*.apps.<cluster_domain>`` to resolve outside of the VPC, link:https://aws.amazon.com/premiumsupport/knowledge-center/route53-resolve-with-inbound-endpoint/[configure a Route53 Resolver Inbound Endpoint].
```
Comment (bmcelvee) [inline modules/osd-aws-privatelink-config-dns-forwarding.adoc:19]: [in the link title] s/Route53/Route 53

File: modules/osd-aws-privatelink-required-resources.adoc [hunk 4]
```diff
@@ -1,7 +1,7 @@
 [id="osd-aws-privatelink-required-resources.adoc_{context}"]
-= Requirements for using PrivateLink clusters
+= Requirements for using AWS PrivateLink clusters
 
-For PrivateLink clusters, internet gateways, NAT gateways and public subnets are not required, but the private subnets must have internet connectivity provided in order to install required components. At least one single private subnet is required for Single-AZ clusters and at least 3 private subnets are required for Multi-AZ clusters. The following table shows the AWS resources that are required for a successful installation:
+For AWS PrivateLink clusters, internet gateways, NAT gateways and public subnets are not required, but the private subnets must have internet connectivity provided in order to install required components. At least one single private subnet is required for Single-AZ clusters and at least 3 private subnets are required for Multi-AZ clusters. The following table shows the AWS resources that are required for a successful installation:
```
Comment (bmcelvee) [inline modules/osd-aws-privatelink-required-resources.adoc:4]: s/in order to/to

**Comment by bmcelvee** [comment]
Just a couple of edits, otherwise LGTM!

A few other notes: 

1. Don't forget to squash your commits. :smile_cat: 
2. And this is outside of the scope of this PR, but I noticed a rouge "+" in the _Creating an AWS PrivateLink cluster_ assembly in the preview: https://deploy-preview-35314--osdocs.netlify.app/openshift-rosa/latest/rosa_getting_started/rosa-aws-privatelink-creating-cluster?utm_source=github&utm_campaign=bot_dp#rosa-aws-privatelink-create-cluster.adoc_rosa-getting-started

**Comment by EricPonvelle** [comment]
@arendej can you take a look at this PR for SME acknowledgement?

**Comment by arendej** [comment]
I'm good with this one and agree with @bmcelvee above.

**Pull request merged**